src/app/components/product/product.component.spec.ts

```diff
@@ -36,28 +35,35 @@ describe('AppComponent => ProductComponent', () => {
       })
       .compileComponents();
 
-    fixture = TestBed.createComponent(ProductComponent);
+    fixture = TestBed.createComponent(TestComponent);
+    loader = TestbedHarnessEnvironment.loader(fixture);
     component = fixture.componentInstance;
 
     component.product = product;
     fixture.detectChanges();
+    await fixture.whenStable();
   });
 
   it('should be created', () => {
     expect(component).not.toBeNull();
   });
 
-  it('should show product name', () => {
-    expect(Select.title.nativeElement.innerText).toBe(product.name);
+  it('should show product name', async () => {
+    const productCom = await loader.getHarness(ProductHarness);
```

Comment:
Please choose more appropriate name, e.g. *productHarness* or just *harness*.

The same applies to the code below.